### PR TITLE
fix: Plausible update, analytics updates for the Share Public Link feature #533

### LINKED ISSUE #533: Landscapes map show empty map if longitude out of range
## Description
- Landscapes map not showing anything if a landscape longitude value is out of range

## Steps To Reproduce

1. Create or update a landscape with a longitude value outside of `[-180, 180]` in this case `612616.230011`
2. Open /landscape page

## Expected behavior
- Show all landscape markers

## Actual behavior
Not showing any landscape

### PR DESCRIPTION
**Pull request by tm-ruxandra**
## Description

Address requests from [Plausible update, analytics updates for the Share Public Link feature #533](https://github.com/techmatters/terraso-product/issues/533). Both download buttons now generate an event, with common code ensuring consistency between the two. Additionally, 'share public link' functionality now uses standardized events as well. Both updated plausible events now live in a common `events.js` file so we can begin centralizing Plausible-related logic with an eye towards having centralized control over all events.

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added

### Related Issues
Fixes #533 

### Verification steps

Run unit tests where added. Go to downloads for both a group and landscape, press the download button, and verify in the network inspector that the outgoing plausible event calls have the expected parameters.

### CODE REVIEW

File: src/monitoring/events.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2021-2024 Technology Matters
```
Comment (paulschreiber) [inline src/monitoring/events.js:2]:
```suggestion
 * Copyright © 2024 Technology Matters
```

File: src/monitoring/events.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2021-2023 Technology Matters
```
Comment (paulschreiber) [inline src/monitoring/events.test.js:2]:
```suggestion
 * Copyright © 2024 Technology Matters
```

File: src/monitoring/events.test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      url: expectedUrl,
+    },
+  });
+});
+
+test('Events: onDownload', async () => {
+  const Component = () => {
+    const { onDownload } = useDownloadEvent();
+    onDownload('group', 'abc', 'test');
+    return <div></div>;
```
Comment (paulschreiber) [inline src/monitoring/events.test.js:57]:
```suggestion
    return <div />;
```

File: src/monitoring/events.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+export const useDownloadEvent = () => {
+  const { trackEvent } = useAnalytics();
+
+  const onDownload = useCallback(
+    (entityType, ownerSlug, location) => {
+      trackEvent('dataEntry.file.download', {
+        props: {
+          [entityType]: ownerSlug,
+          'download location': location,
```
Comment (paulschreiber) [inline src/monitoring/events.js:44]:
```suggestion
          'downloadLocation': location,
```

**Review by paulschreiber — APPROVED** [review]
Looks good. Left a few small fixes in comments.

**Comment by paulschreiber** [comment]
@tm-ruxandra Note that the PR title needs to have a conventional commits prefix. I updated it for you.

File: src/monitoring/events.js (hunk 4 above)
Comment (tm-ruxandra) [reply src/monitoring/events.js:44]: This was specifically requested in the originating issue - I can confirm with Derek whether this is okay. I definitely agree that having a conventional property name would be better here.

**Pull request merged**